asv_bench/benchmarks/benchmarks_hindcast.py

```diff
@@ -0,0 +1,166 @@
+        self.hist = xr.Dataset()
+
+        self.nmember = 3
+        self.nlead = 3
+        self.nx = 64
+        self.ny = 64
+        self.init_start = 1960
+        self.init_end = 2000
+        self.ninit = self.init_end - self.init_start
+
+        frac_nan = 0.0
+
+        inits = np.arange(self.init_start, self.init_end)
+        leads = np.arange(1, 1 + self.nlead)
+        members = np.arange(1, 1 + self.nmember)
+
+        lons = xr.DataArray(
+            np.linspace(0, 360, self.nx),
+            dims=('lon',),
+            attrs={'units': 'degrees east', 'long_name': 'longitude'},
+        )
+        lats = xr.DataArray(
+            np.linspace(-90, 90, self.ny),
+            dims=('lat',),
+            attrs={'units': 'degrees north', 'long_name': 'latitude'},
+        )
+        self.hind['tos'] = xr.DataArray(
+            randn(
+                (self.nmember, self.ninit, self.nlead, self.nx, self.ny),
+                frac_nan=frac_nan,
+            ),
+            coords={
+                'member': members,
+                'init': inits,
+                'lon': lons,
+                'lat': lats,
+                'lead': leads,
+            },
+            dims=('member', 'init', 'lead', 'lon', 'lat'),
+            name='tos',
+            encoding=None,
+            attrs={'units': 'foo units', 'description': 'a description'},
+        )
+        self.reference['tos'] = xr.DataArray(
+            randn((self.ninit, self.nx, self.ny), frac_nan=frac_nan),
+            coords={'lon': lons, 'lat': lats, 'time': inits},
+            dims=('time', 'lon', 'lat'),
+            name='tos',
+            encoding=None,
+            attrs={'units': 'foo units', 'description': 'a description'},
+        )
+
+        self.hist['tos'] = xr.DataArray(
+            randn((self.ninit, self.nx, self.ny, self.nmember), frac_nan=frac_nan),
+            coords={'lon': lons, 'lat': lats, 'time': inits, 'member': members},
+            dims=('time', 'lon', 'lat', 'member'),
+            name='tos',
+            encoding=None,
+            attrs={'units': 'foo units', 'description': 'a description'},
+        )
+
+        self.hind.attrs = {'history': 'created for xarray benchmarking'}
+
+
+class Compute(Generate):
+    """
+    A few examples that benchmark climpred compute_hindcast.
+    """
+
+    def setup(self, *args, **kwargs):
+        self.make_hind_ref()
+
+    @parameterized(['metric', 'comparison'], (METRICS, HINDCAST_COMPARISONS))
+    def time_compute_hindcast(self, metric, comparison):
+        """Take time for compute_hindcast."""
+        _ensure_loaded(
+            compute_hindcast(
+                self.hind, self.reference, metric=metric, comparison=comparison
+            )
+        )
+
+    @parameterized(['metric', 'comparison'], (METRICS, HINDCAST_COMPARISONS))
+    def peakmem_compute_hindcast(self, metric, comparison):
+        """Take memory peak for compute_hindcast for all comparisons."""
```

Comment:
This is missing `e2o`. Seems like it's just `m2o` right now.

Reply:
I don't want to measure all comparisons and metrics combinations. also `e2o` fails with probabilistic metric